src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/CosmosDb/QueryIteratorsIntegrations.cs

```diff
@@ -0,0 +1,308 @@
+
+                var parent = tracer.ActiveScope?.Span;
+
+                if (parent != null &&
+                    parent.Type == SpanTypes.CosmosDb &&
+                    parent.ResourceName == query)
+                {
+                    // we are already instrumenting this,
+                    // don't instrument nested methods that belong to the same stacktrace
+                    return new CallTargetState(null);
+                }
+
+                string containerId = null;
+                string databaseId = null;
+                string endpoint = null;
+
+                object databaseObj = null;
+                object clientObj = null;
+
+                if (DuckType.CanCreate(typeof(IContainerOld), instance))
+                {
+                    var container = DuckType.Create<IContainerOld>(instance);
+                    containerId = container.Id;
+                    databaseObj = container.Database;
+                }
+                else if (DuckType.CanCreate(typeof(IContainerNew), instance))
+                {
+                    var container = DuckType.Create<IContainerNew>(instance);
+                    containerId = container.Id;
+                    databaseObj = container.Database;
+                }
+                else
+                {
+                    databaseObj = instance;
+                }
+
+                if (DuckType.CanCreate(typeof(IDatabaseOld), databaseObj))
+                {
+                    var database = DuckType.Create<IDatabaseOld>(databaseObj);
+                    databaseId = database.Id;
+                    clientObj = database.ClientContext.Client;
+                }
+                else if (DuckType.CanCreate(typeof(IDatabaseNew), databaseObj))
+                {
+                    var database = DuckType.Create<IDatabaseNew>(databaseObj);
+                    databaseId = database.Id;
+                    endpoint = database.Client.Endpoint.ToString();
+                    clientObj = database.Client;
+                }
+                else
+                {
+                    clientObj = instance;
+                }
+
+                if (DuckType.CanCreate(typeof(ICosmosClient), clientObj))
+                {
+                    var database = DuckType.Create<ICosmosClient>(clientObj);
+                    endpoint = database.Endpoint.ToString();
+                }
+
+                var tags = new CosmosDbTags
+                {
+                    ContainerId = containerId,
+                    DatabaseId = databaseId,
+                    Host = endpoint
+                };
+
+                var serviceName = tracer.Settings.GetServiceName(tracer, ServiceName);
+                var scope = tracer.StartActiveWithTags(OperationName, tags: tags, serviceName: serviceName);
+
+                var span = scope.Span;
+
+                span.ResourceName = query;
+                span.Type = SpanTypes.CosmosDb;
```

Comment:
If resource names look like SQL queries (`SELECT X FROM Y`), we should consider using the `"sql"` span type (and add a tag `db.type = cosmosdb`). The span type will trigger query obfuscating in the Agent and special features in the UI, like syntax highlighting and showing the services under the "db" category.

Reply:
Did we make any change here? If not, can you explain? Thanks. (unresolving)